test/tests/middlewares/request-unflattener.test.js

```diff
@@ -0,0 +1,288 @@
+const requestUnflattener = require('../../../src/middlewares/request-unflattener');
```

Comment:
I've not looked the tests into details, but the three major stuffs to test are present: 
- Unflatten attributes
- Unflatten requested fields in query
- Unflatten requested fields in sub query (select all)